functions/server/fn_preparePlaybackServer.sqf

```diff
@@ -2,20 +2,22 @@
 
 if (!isServer) exitWith {};
 
-{
-	if (isMultiplayer) then {
-		_x enableSimulationGlobal false;
-
-		// freeze vehicle
-		if (!(vehicle _x isEqualTo _x) && _x isEqualTo (driver vehicle _x)) then {
-				vehicle _x attachTo [_x];
-		};
-	} else {
-		_x enableSimulation false;
-	};
+private _fnc_disableUnit = if (isMultiplayer) then {
+    {
+        _this enableSimulationGlobal false;
+
+        // freeze vehicle
+        if (!(vehicle _this isEqualTo _this) && _this isEqualTo (driver vehicle _this)) then {
+            vehicle _this attachTo [_this];
+        };
+    }
+} else {
+    {_this enableSimulation false}
+};
 
+{
+    _x call _fnc_disableUnit;
```

Comment:
If you do
if isMultiplayer then
<first forEachLoop variant>
else
<second variant>

That'll be faster because you spare yourself the `call`. Yes it'll be more code and also a bit of duplicate code.

Reply:
Also you can use `count` for the loop here. As setVariable always returns `nil` which is valid return value for `count`

Reply:
That's fine I guess, this only runs once. I'll take fewer lines of code over the performance.